lib/sisimai/bite/email/facebook.rb

```diff
@@ -0,0 +1,274 @@
+        ],
+        rejected: [
+          'DNS-P1', # Your SMTP MAIL FROM domain does not exist.
+          'DNS-P2', # Your SMTP MAIL FROM domain does not have an MX record.
+          'DNS-T1', # Your SMTP MAIL FROM domain exists but does not currently resolve.
+          'DNS-P3', # Your mail server does not have a reverse DNS record.
+          'DNS-T2', # You mail server's reverse DNS record does not currently resolve.
+        ],
+        systemerror: [
+          'CON-T1', # Facebook's mail server currently has too many connections open to allow another one.
+        ],
+        toomanyconn: [
+          'CON-T3', # Your mail server has opened too many new connections to Facebook's mail servers in a short period of time.
+        ],
+        suspend: [
+          'RCP-T4', # The attempted recipient address is currently deactivated. The user may or may not reactivate it.
+        ],
+        undefined: [
+          'RCP-T1', # The attempted recipient address is not currently available due to an internal system issue. This is a temporary condition.
+          'MSG-T1', # The number of recipients on the message exceeds Facebook's allowed maximum.
+          'CON-T2', # Your mail server currently has too many connections open to Facebook's mail servers.
+          'CON-T4', # Your mail server has exceeded the maximum number of recipients for its current connection.
+        ],
+      }
+      Indicators = Sisimai::Bite::Email.INDICATORS
+
+      def description; return 'Facebook: https://www.facebook.com'; end
+      def smtpagent;   return Sisimai::Bite.smtpagent(self); end
+      def headerlist;  return []; end
+      def pattern;     return Re0; end
+
+      # Parse bounce messages from Facebook
+      # @param         [Hash] mhead       Message headers of a bounce email
+      # @options mhead [String] from      From header
+      # @options mhead [String] date      Date header
+      # @options mhead [String] subject   Subject header
+      # @options mhead [Array]  received  Received headers
+      # @options mhead [String] others    Other required headers
+      # @param         [String] mbody     Message body of a bounce email
+      # @return        [Hash, Nil]        Bounce data list and message/rfc822
+      #                                   part or nil if it failed to parse or
+      #                                   the arguments are missing
+      def scan(mhead, mbody)
+        return nil unless mhead
+        return nil unless mbody
+        return nil unless mhead['from']    =~ Re0[:from]
+        return nil unless mhead['subject'] =~ Re0[:subject]
+
+        dscontents = [Sisimai::Bite.DELIVERYSTATUS]
+        hasdivided = mbody.split("\n")
+        havepassed = ['']
+        rfc822list = []     # (Array) Each line in message/rfc822 part string
+        blanklines = 0      # (Integer) The number of blank lines
+        readcursor = 0      # (Integer) Points the current cursor position
```

Comment:
__[rubocop]__
* Metrics/AbcSize: Assignment Branch Condition size for scan is too high. [120.4/30] [:ref](http://c2.com/cgi/wiki?AbcMetric)

<!-- Comment by [SideCI](https://sideci.com) -->